### PR TITLE
refactor(nodebuilder): use external flock library

### PR DESCRIPTION
**Pull request by Wondertan**
This PR is prompted by #3246. While looking through the code and understanding how `IsLocked` worked there, I realized that our fslock lib was missing a feature. Instead of asking @mastergaurang94 to fix it, I decided to completely migrate over to a more [comprehensive locking solution](https://github.com/gofrs/flock), which additionally:
* avoids the need custom code to maintain(NIHS)
* solves the problem of locking being annoying after an ungraceful shutdown
* gives us Windows and other niche OS support
	* as well as works around niche FS edge-cases

### CODE REVIEW

**Review by ramin — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]
LGTM. General nit:` f.Close` calls `Unlock` under the hood, so you can unify the code by choosing only one method

File: nodebuilder/init_test.go [hunk 1]
```diff
@@ -58,9 +58,10 @@ func TestIsInitForNonExistDir(t *testing.T) {
 
 func TestInitErrForLockedDir(t *testing.T) {
 	dir := t.TempDir()
-	flock, err := fslock.Lock(lockPath(dir))
+	flk := flock.New(lockPath(dir))
+	_, err := flk.TryLock()
 	require.NoError(t, err)
-	defer flock.Unlock() //nolint:errcheck
+	defer flk.Unlock() //nolint:errcheck
```
Comment (vgonkivs) [inline nodebuilder/init_test.go:64]: could be changed to `flk.Close()`

File: nodebuilder/store.go [hunk 2]
```diff
@@ -58,28 +58,29 @@ func OpenStore(path string, ring keyring.Keyring) (Store, error) {
...
+	}
+
+	if !IsInit(path) {
+		err := errors.Join(ErrNotInited, flk.Unlock())
+		return nil, err
 	}
 
 	ks, err := keystore.NewFSKeystore(keysPath(path), ring)
 	if err != nil {
+		err = errors.Join(err, flk.Unlock())
```
Comment (vgonkivs) [inline nodebuilder/store.go:77]: could be changed to `flk.Close()`

File: nodebuilder/store.go [hunk 3]
```diff
@@ -58,28 +58,29 @@ func OpenStore(path string, ring keyring.Keyring) (Store, error) {
...
-
-	ok := IsInit(path)
 	if !ok {
-		flock.Unlock() //nolint:errcheck
-		return nil, ErrNotInited
+		return nil, ErrOpened
+	}
+
+	if !IsInit(path) {
+		err := errors.Join(ErrNotInited, flk.Unlock())
```
Comment (vgonkivs) [inline nodebuilder/store.go:71]: could be changed to `flk.Close()`

**Review by vgonkivs — COMMENTED** [review]
Still, it's a nit and it's up to you

**Review by vgonkivs — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**